api/seek_stats.go

```diff
@@ -0,0 +1,64 @@
+package api
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/joschahenningsen/TUM-Live/dao"
+	log "github.com/sirupsen/logrus"
+	"net/http"
+)
+
+// progressRoutes contains a DaoWrapper object and all route functions dangle from it.
+type seekStatsRoutes struct {
+	dao.DaoWrapper
+}
+
+// configSeekStatsRouter sets up the router and initializes a progress buffer
+// that is used to minimize writes to the database.
+func configSeekStatsRouter(router *gin.Engine, daoWrapper dao.DaoWrapper) {
+	routes := seekStatsRoutes{daoWrapper}
+
+	router.POST("/api/seekReport/:streamID", routes.reportSeek)
```

Comment:
Currently anyone can `post` reports. There is no "firewall" so to say. I don't know how problematic that is but malign folk might abuse that behavior.

A naive solution would be to only report logged-in seeks. That, however, potentially leads to fewer data points.

Otherwise we could install some sort of "spam-protection". This should be configurable in the nginx instance TUM-Live uses or maybe already is. There might even be some `gin-gonic` features that can deal with that problem.

Reply:
Yes already thought about this in the beginning.

We wanted to store no userdata related to the seeks, we could store users who trigger the endpoint separately from the seek data, so there is no link between the user and the seek-data but still we would kind of capture the users who trigger the endpoint somehow... Idk :)

We could limit invocation by IP in a certain time frame. Probably better to set that on nginx level ? Not sure :)